src/pages/home/report/ReactionList/PopoverReactionList.js

```diff
@@ -198,5 +317,15 @@ class PopoverReactionList extends React.Component {
 }
 
 PopoverReactionList.propTypes = propTypes;
+PopoverReactionList.defaultProps = defaultProps;
 
-export default withLocalize(PopoverReactionList);
+export default compose(
+    withLocalize,
+    withOnyx({
+        reportActions: {
+            key: ({reportID}) => `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
+            canEvict: false,
```

Comment:
Why is this here?

Reply:
@rushatgabhane  The PR is merged, so do we create a new PR to fix it or revert this PR?

Reply:
This is an explanation about canEvict field
https://github.com/Expensify/react-native-onyx/blob/1d496b001f327c509131597faf260a81c002be77/lib/withOnyx.js#L113-L116
@tgolen @rushatgabhane  I see that `canEvict: false` is added when we get reportAction everywhere in App repo. So I think in here we also add it for consistency

Reply:
I'm pretty sure the eviction logic is completely broken, and I think this should just be removed, but maybe it should be there for report actions... I do see it used everywhere. @vitHoracek @marcaaron do you have any guidance on the `canEvict` setting for report actions?

Reply:
Ok, I did some poking and all the places where it is being used look kosher to me. There is documented guidance here: https://github.com/expensify/app#storage-eviction

Open to improvements or suggestions. I think there's _probably_ a better version of storage eviction waiting to be built at some point in all of our imaginations - but this one has gonna us pretty far. Let me know if there's anything else I can help with 🙇‍♂️

Reply:
@marcaaron The reason I believe that the eviction logic is broken is from this line here: https://github.com/Expensify/react-native-onyx/blob/main/lib/withOnyx.js#L129

When I was working with Thaibault (during ExpensiconX) and looking at migrating Onxy to TypeScript, we found that `mapping.connectionID` is `undefined`. This is because:

- The `connectionID` gets saved to the mapping [here](https://github.com/Expensify/react-native-onyx/blob/main/lib/Onyx.js#L676) in `Onyx.connect()`
- We assume that the `mapping` there is a reference to the same mapping object that is in `withOnyx` but actually, it's a copy as we can see by the spread operator being used on `mapping` [here](https://github.com/Expensify/react-native-onyx/blob/main/lib/withOnyx.js#L151) when it is passed to `Onyx.connect()`
- `checkEvictableKeys()` then looks through all the mappings in `mapOnyxToState` [here](https://github.com/Expensify/react-native-onyx/blob/main/lib/withOnyx.js#LL116C24-L116C38) and references `mapping.connectionID`. This is where it breaks 💥 because that object never had `connectionID` saved to it, the `connectionID` was only saved to a copy of it.